### PR TITLE
Add --reload-dir command line argument

### PR DESCRIPTION
**Pull request by jcheng5**
Also, setting env var SHINY_PKG_AUTORELOAD=1 causes autoreload on changes to the Shiny package itself (at least with `pip install -e`, which is how I'm using it).

### CODE REVIEW

File: shiny/_main.py [hunk 1]
```diff
@@ -249,6 +266,7 @@ def run_app(
         debug=debug,
         reload=reload,
         reload_dirs=reload_dirs,
+        reload_includes=["*.py", "*.css", "*.js", "*.htm", "*.html", "*.png"],
```
Comment (cpsievert) [inline shiny/_main.py:269]: Maybe worth including svg and json?

File: shiny/_main.py [hunk 2]
```diff
@@ -220,12 +232,17 @@ def run_app(
...
         reload_dirs = []
-
     if reload:
+        # Always watch the app_dir
+        if app_dir:
+            reload_dirs.append(app_dir)
+        # For developers of Shiny itself; autoreload the app when Shiny package changes
+        if os.getenv("SHINY_PKG_AUTORELOAD"):
+            shinypath = Path(inspect.getfile(shiny)).parent
+            reload_dirs.append(str(shinypath))
```
Comment (cpsievert) [inline shiny/_main.py:244]: Super neat! So are you `export`ing this in your `.profile`, or something better?

**Review by cpsievert — APPROVED** [review]

File: shiny/_main.py (hunk 1 above)
Comment (jcheng5) [reply shiny/_main.py:269]:
```suggestion
        reload_includes=["*.py", "*.css", "*.js", "*.htm", "*.html", "*.png", "*.svg", "*.json"],
```

File: shiny/_main.py (hunk 2 above)
Comment (jcheng5) [reply shiny/_main.py:244]: Just in .profile.

**Comment by nealrichardson** [comment]
@jcheng5 Is this PR still desired? Happy to rebase, resolve conflicts, and merge if so.

**Pull request merged**